### PR TITLE
various: move path and I/O utility functions to separate files

### PR DESCRIPTION
**Pull request by na-na-hi**
These utility functions are functionally independent from the files they live in (which deal with config path and Windows I/O respectively). As a result, the include paths for these functions are very misleading. 

This improves organization of these functions by separating them to two separate files, `misc/io_utils.c` and `misc/path_utils.c`, to emphasize the fact that they're purely utility functions.

### CODE REVIEW

File: misc/path_utils.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with mpv.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef MPLAYER_PATH_UTIL_H
+#define MPLAYER_PATH_UTIL_H
```
Comment (sfan5) [inline misc/path_utils.h:21]: `#pragma once` for new code

**Comment by sfan5** [comment]
I think the `_utils.c` naming is a bit ugly.
How about `misc/{io,path}.{c,h}`? Or do you think that would be confusing?

**Review by sfan5 — APPROVED** [review]

File: misc/io_utils.c [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+#include <limits.h>
+#include <unistd.h>
+
+#include "mpv_talloc.h"
+#include "config.h"
+#include "misc/random.h"
+#include "misc/io_utils.h"
+#include "osdep/io.h"
+
+int mp_mkostemps(char *template, int suffixlen, int flags)
```
Comment (kasper93) [inline misc/io_utils.c:34]: I know this is only anime media player, but this function triggers me so much.

Comment (sfan5) [reply misc/io_utils.c:34]: do you want to blame GNU for this?
https://man.archlinux.org/man/mkostemps.3

Comment (kasper93) [reply misc/io_utils.c:34]: Blame what you want, I don't see any reason for those variable names in this function.

Comment (sfan5) [reply misc/io_utils.c:34]: well that's just wm4

Comment (kasper93) [reply misc/io_utils.c:34]: So that's why you want to be on the git blame for it. Fair.

**Pull request merged**